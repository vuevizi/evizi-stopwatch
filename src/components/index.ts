import Watch from "./Watch/Watch";

export {Watch}