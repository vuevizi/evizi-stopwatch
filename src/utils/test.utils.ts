export const findTestElement = (component: any, attr: string) => {
  const findElement: any = component.find(`[data-test="${attr}"]`);
  return findElement;
};
