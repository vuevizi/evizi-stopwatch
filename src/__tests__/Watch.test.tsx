import {Watch} from '../components/';
import {mount} from 'enzyme';
import configureStore from 'redux-mock-store';
import {Middleware} from '@reduxjs/toolkit';
import {Provider} from 'react-redux';
import {findTestElement} from '../utils/test.utils';
import toJson from 'enzyme-to-json';

const middlewares: Middleware[] = [];
const mockStore = configureStore(middlewares);

describe('Watch Test', () => {
  let store;
  let component: any;
  let watchContainer: any;
  let watch: any;
  let watchActions: any;
  beforeEach(() => {
    store = mockStore({
      amountOfRecords: 0,
      timeRecords: [],
    });
    component = mount(
      <Provider store={store}>
        <Watch />
      </Provider>
    );
    watch = findTestElement(component, 'watch');
    watchContainer = findTestElement(component, 'watch-container');
    watchActions = findTestElement(component, 'watch-actions');
  });
  test('renders without crash', () => {
    expect(component.length).toBe(1);
  });
  test('renders matching with snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
  test('renders right minutes,seconds,milliseconds elements', () => {
    const minutesText = findTestElement(watch, 'minutes');
    const secondsText = findTestElement(watch, 'seconds');
    const millisecondsText = findTestElement(watch, 'milliseconds');
    expect(millisecondsText.length).toEqual(1);
    expect(minutesText.length).toEqual(1);
    expect(secondsText.length).toEqual(1);
    expect(minutesText.text()).toEqual('00');
    expect(secondsText.text()).toEqual('00');
    expect(millisecondsText.text()).toEqual('00');
  });
  test('renders right button Start at the first render', () => {
    const buttonStart = findTestElement(watchActions, 'button-start');
    expect(buttonStart.length).toBe(1);
    console.log(buttonStart.instance());
  });
  test('renders right button Stop at the first render', () => {
    const buttonStop = findTestElement(watchActions, 'button-stop');
    expect(buttonStop.length).toBe(0);
  });
  test('renders right button Mark at the first render', () => {
    const buttonMark = findTestElement(watchActions, 'button-mark');
    expect(buttonMark.length).toBe(0);
  });
  test('renders right button Reset at the first render', () => {
    const buttonReset = findTestElement(watchActions, 'button-reset');
    expect(buttonReset.length).toBe(1);
  });
});
