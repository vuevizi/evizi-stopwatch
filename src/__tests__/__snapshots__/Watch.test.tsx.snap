// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Watch Test renders matching with snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Watch>
    <styled.div
      data-test="watch-container"
    >
      <div
        className="sc-bcXHqe kRtKIC"
        data-test="watch-container"
      >
        <styled.div
          data-test="watch"
        >
          <div
            className="sc-gswNZR iotAsC"
            data-test="watch"
          >
            <div
              className="watch__minutes-container"
            >
              <h1
                data-test="minutes"
                id="minutes"
              >
                00
              </h1>
            </div>
            :
            <div
              className="watch__seconds-container"
            >
              <h1
                data-test="seconds"
                id="seconds"
              >
                00
              </h1>
            </div>
            :
            <div
              className="watch__milliseconds-container"
            >
              <h1
                data-test="milliseconds"
                id="milliseconds"
              >
                00
              </h1>
            </div>
          </div>
        </styled.div>
        <styled.div
          data-test="watch-actions"
        >
          <div
            className="sc-dkrFOg iSQxeL"
            data-test="watch-actions"
          >
            <styled.button
              className="button-start"
              data-test="button-start"
              onClick={[Function]}
            >
              <button
                className="sc-hLBbgP gavrQs button-start"
                data-test="button-start"
                onClick={[Function]}
              >
                Start
              </button>
            </styled.button>
            <styled.button
              data-test="button-reset"
              onClick={[Function]}
            >
              <button
                className="sc-jSUZER fahfkc"
                data-test="button-reset"
                onClick={[Function]}
              >
                Reset
              </button>
            </styled.button>
          </div>
        </styled.div>
      </div>
    </styled.div>
  </Watch>
</Provider>
`;
